packages/core/streams/openai-stream.ts

```diff
@@ -51,6 +53,59 @@ export type OpenAIStreamCallbacks = AIStreamCallbacks & {
   ) => Promise<Response | undefined | void | string>
 }
 
+// https://github.com/openai/openai-node/blob/07b3504e1c40fd929f4aae1651b83afc19e3baf8/src/resources/chat/completions.ts#L28-L40
+interface ChatCompletionChunk {
+  id: string
+  choices: Array<ChatCompletionChunkChoice>
+  created: number
+  model: string
+  object: string
+}
+
+// https://github.com/openai/openai-node/blob/07b3504e1c40fd929f4aae1651b83afc19e3baf8/src/resources/chat/completions.ts#L43-L49
+interface ChatCompletionChunkChoice {
+  delta: ChoiceDelta
+
+  finish_reason: 'stop' | 'length' | 'function_call' | null
+
+  index: number
+}
+
+// https://github.com/openai/openai-node/blob/07b3504e1c40fd929f4aae1651b83afc19e3baf8/src/resources/chat/completions.ts#L123-L139
+interface ChoiceDelta {
+  /**
+   * The contents of the chunk message.
+   */
+  content?: string | null
+
+  /**
+   * The name and arguments of a function that should be called, as generated by the
+   * model.
+   */
+  function_call?: FunctionCall
+
+  /**
+   * The role of the author of this message.
+   */
+  role?: 'system' | 'user' | 'assistant' | 'function'
+}
+
+// https://github.com/openai/openai-node/blob/07b3504e1c40fd929f4aae1651b83afc19e3baf8/src/resources/chat/completions.ts#L146-L159
+interface FunctionCall {
```

Comment:
nit: we use `ChatCompletionRequestMessageFunctionCall` from `openai-edge` elsewhere in the library. I'm happy keeping the types declared here but wondering if we want to unify them.

Reply:
Ideally we'd delete all of these when we can import the `ChatCompletionChunk` type.